src/Nethermind/Nethermind.Core/Caching/ClockCache.cs

```diff
@@ -0,0 +1,248 @@
+            {
+                _keys.AsSpan().Clear();
+            }
+            _accessedBitmap.AsSpan().Clear();
+        }
+
+        public bool Contains(TKey key)
+        {
+            return _cacheMap.ContainsKey(key);
+        }
+
+        public int Count => _cacheMap.Count;
+
+        private bool ClearAccessed(int position)
+        {
+            uint offset = (uint)position >> BitShiftPerInt64;
+            long flags = 1L << position;
+
+            ref long accessedBitmapWord = ref Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(_accessedBitmap), offset);
+            bool accessed = (accessedBitmapWord & flags) != 0;
+
+            if (accessed)
+            {
+                // Clear the accessed bit
+                flags = ~flags;
+                long current = Volatile.Read(ref accessedBitmapWord);
+                while (true)
+                {
+                    long previous = Interlocked.CompareExchange(ref accessedBitmapWord, current & flags, current);
+                    if (previous == current)
+                    {
+                        break;
+                    }
+                    current = previous;
+                }
+            }
+
+            return accessed;
+        }
+
+        private void MarkAccessed(int position)
+        {
+            uint offset = (uint)position >> BitShiftPerInt64;
+            long flags = 1L << position;
+
+            ref long accessedBitmapWord = ref Unsafe.Add(ref MemoryMarshal.GetArrayDataReference(_accessedBitmap), offset);
+
+            long current = Volatile.Read(ref accessedBitmapWord);
+            while (true)
+            {
+                long previous = Interlocked.CompareExchange(ref accessedBitmapWord, current | flags, current);
+                if (previous == current)
+                {
+                    break;
+                }
+                current = previous;
+            }
+        }
+
+        /// <summary>
+        /// Used for conversion between different representations of bit array.
+        /// Returns (n + (64 - 1)) / 64, rearranged to avoid arithmetic overflow.
+        /// For example, in the bit to int case, the straightforward calc would
+        /// be (n + 63) / 64, but that would cause overflow. So instead it's
+        /// rearranged to ((n - 1) / 64) + 1.
+        /// Due to sign extension, we don't need to special case for n == 0, if we use
+        /// bitwise operations (since ((n - 1) >> 6) + 1 = 0).
+        /// This doesn't hold true for ((n - 1) / 64) + 1, which equals 1.
+        ///
+        /// Usage:
+        /// GetInt32ArrayLengthFromBitLength(77): returns how many ints must be
+        /// allocated to store 77 bits.
+        /// </summary>
+        /// <param name="n"></param>
+        /// <returns>how many ints are required to store n bytes</returns>
+        private static int GetInt64ArrayLengthFromBitLength(int n) =>
+            (n - 1 + (1 << BitShiftPerInt64)) >>> BitShiftPerInt64;
```

Comment:
Make this some helper method available for all caches?

Reply:
Moved to base class